src/YarpUtilities/src/RosPublisher.cpp

```diff
@@ -0,0 +1,482 @@
+}
+
+RosPublisher::~RosPublisher() = default;
+
+bool RosPublisher::initialize(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler)
+{
+    std::string printPrefix{"[RosPublisher::initialize] "};
+    bool ok{true};
+    
+    auto handle = handler.lock();
+    if (handle == nullptr)
+    {
+        std::cerr << printPrefix << "The parameter handler has expired. Please check its scope." << std::endl;
+        return false;
+    }
+
+    // configure joint state publisher if it exists
+    if (handle->getParameter("joint_states_topic", m_pimpl->jointStatePublisher.topic))
+    {
+        if (!configureJointStatePublisher(m_pimpl->jointStatePublisher.topic))
+        {
+            std::cerr << printPrefix << "Could not configure joint states publisher." << std::endl;
+            return false;            
+        }
+    }
+    
+    // configure transform publisher
+    std::string tfServerPort;
+    if (handle->getParameter("transform_server_port", tfServerPort))
+    {
+        m_pimpl->publishTF = true;
+        if (!configureTransformServer(tfServerPort))
+        {
+            return false;            
+        }
+    }
+    
+    auto wrenchHandle = handle->getGroup("WrenchPublishers");
+    auto wrenchHandler = wrenchHandle.lock();    
+    if (wrenchHandler != nullptr)
+    {
+        std::vector<std::string> frames;
+        if (!m_pimpl->setupParamV(wrenchHandler, "frame_names", frames, printPrefix))
+        {
+            return false;
+        }
+        
+        std::vector<std::string> topics;
+        if (!m_pimpl->setupParamV(wrenchHandler, "topics", topics, printPrefix))
+        {
+            return false;
+        }
+        
+        if (frames.size() != topics.size())
+        {
+            std::cerr << printPrefix << " WrenchPublishers group: frame_names and topics must be of same size." << std::endl;
+            return false;              
+        }
+        
+        for (std::size_t idx = 0; idx < frames.size(); idx++)
+        {
+            if (!configureWrenchPublisher(frames[idx], topics[idx]))
+            {
+                std::cerr << printPrefix << "Could not configure wrench publisher for frame: "
+                          << frames[idx] << " and topic: " << topics[idx] << std::endl;
+                return false;            
+            }
+        }
+    }
+                       
+    if (!ok)
+    {   
+        std::cerr << printPrefix << "Failed to initialize the ROSPublisher" << std::endl;
+        return false;
+    }
+
+    m_pimpl->initialized = true;
+
+    return true;
+}
+
+bool RosPublisher::configureJointStatePublisher(const std::string& topicName)
+{
+    if (m_pimpl->jointStatePublisher.ptr == nullptr)
+    {
+        m_pimpl->jointStatePublisher.ptr = std::make_unique< yarp::os::Publisher<JointStateMsg> >();
+    }
+    
+    m_pimpl->jointStatePublisher.ptr->close();
```

Comment:
I have noticed you used this logic a few times. Consider using proper constructors/destructors in order to properly allocate and close these objects automatically.

Reply:
I tried it actually. Don't remember why now, the issue I faced with that. I think I will give it one more try.

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/156/commits/d57a286f4780f87c335ff0b3b91ca01443ffa069